subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DependencyServices.java

```diff
@@ -56,8 +57,8 @@ ArtifactTransformListener createArtifactTransformListener(ListenerManager listen
             return listenerManager.getBroadcaster(ArtifactTransformListener.class);
         }
 
-        TransformationNodeFactory createTransformationNodeFactory() {
-            return new DefaultTransformationNodeFactory();
+        TransformationNodeFactory createTransformationNodeFactory(GradleInternal gradle) {
+            return new DefaultTransformationNodeFactory(gradle);
```

Comment:
I know it's outside the scope of this PR, but I wish we had a way to not expose the world to this service here. :/ All we care about is the identity path.

One thing we could do already to reduce the damage is to pass a `Factory<Path>` instead.

Reply:
Nevermind, I just added another commit to this PR: a382acca34f.